drivers/modem/hl7800.c

```diff
@@ -3117,7 +3115,7 @@ static inline struct net_buf *read_rx_allocator(k_timeout_t timeout,
 
 static size_t hl7800_read_rx(struct net_buf **buf)
 {
-	uint8_t uart_buffer[MDM_RECV_BUF_SIZE];
+	uint8_t uart_buffer[CONFIG_MODEM_HL7800_RECV_BUF_SIZE];
```

Comment:
BTW, you could avoid this and possible stack issues by reading data directly to net_buf. Having this intermediate stack buffer is not really needed.

Reply:
Looked at this and I am not sure how to accomplish it. `mdm_receiver_recv` needs to read to a buffer and then that buffer is used for `net_buf_append_bytes`. I don't think it's possible to do this another way without significant changes.

Reply:
There was a patch few days ago that did similar things (avoiding extra temp buffer read), see commit 318cbe649aed5c2a28dea3ceaff7ccd2af871ab4 for details how it did it.

Reply:
perfect, thank you. I have made note of this and will look at improving this in a new PR.